### PR TITLE
feat: use another webrequest tool 

### PR DESCRIPTION
**Pull request by eordano**

### CODE REVIEW

**Review by pravusjif — APPROVED** [review]
Looks good to me and the improvement is noticable (a lot less stuttering), however I don't understand how we are using this Kongregate's optimization to take advantage of IndexedDB, does this plugin "automagically" take care of that? since we are still using `UnityWebRequest` for the fetching

**Comment by pravusjif** [comment]
> Looks good to me and the improvement is noticable (a lot less stuttering), however I don't understand how we are using this Kongregate's optimization to take advantage of IndexedDB, does this plugin "automagically" take care of that? since we are still using `UnityWebRequest` for the fetching

Just checked the readme in more detail and it says basically that XD

**Review by BrianAmadori — APPROVED** [review]

File: .circleci/hotpatch.sh [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+mv $TARGET $TARGET.gz
+gunzip $TARGET.gz
+
+export LOCATION_START_CUT=282
+export LOCATION_END_CUT=6975
+export FILE_LENGTH=`wc -c $TARGET | grep -oE '[0-9]+'`
+export REST_FROM_BOTTOM=`echo "$FILE_LENGTH-$LOCATION_END_CUT" | bc`
+export CUTTING=`echo "$LOCATION_END_CUT-$LOCATION_START_CUT" | bc`
+
+echo "Cutting:"
```
Comment (BrianAmadori) [inline .circleci/hotpatch.sh:14]: Maybe we can remove this `echo` call

File: .circleci/hotpatch.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+cat $TARGET | head -c $LOCATION_END_CUT | tail -c $CUTTING
+
+head -c $LOCATION_START_CUT $TARGET > $TARGET.new
+tail -c $REST_FROM_BOTTOM $TARGET >> $TARGET.new
+
+mv $TARGET.new $TARGET
+# gzip -n -1 $TARGET
```
Comment (BrianAmadori) [inline .circleci/hotpatch.sh]: Maybe we can remove those comments

File: unity-client/Assets/Plugins/WebGLCachedXHRExtensions/CachedXHRExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,56 @@
+﻿using UnityEngine;
+using System.Runtime.InteropServices;
+
+namespace Kongregate {
+    public class CacheEntryQuery : CustomYieldInstruction {
```
Comment (BrianAmadori) [inline unity-client/Assets/Plugins/WebGLCachedXHRExtensions/CachedXHRExtensions.cs:5]: We should fix the formatting to use our convention.

File: .circleci/hotpatch.sh (hunk 1 above)
Comment (eordano) [reply .circleci/hotpatch.sh:14]: It's pretty useful to have this in CircleCI if this eventually breaks...

File: .circleci/hotpatch.sh (hunk 2 above)
Comment (eordano) [reply .circleci/hotpatch.sh]: Done
```suggestion
```

File: .circleci/hotpatch.sh [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+export CUTTING=`echo "$LOCATION_END_CUT-$LOCATION_START_CUT" | bc`
+
+echo "Cutting:"
+cat $TARGET | head -c $LOCATION_END_CUT | tail -c $CUTTING
+
+head -c $LOCATION_START_CUT $TARGET > $TARGET.new
+tail -c $REST_FROM_BOTTOM $TARGET >> $TARGET.new
+
+mv $TARGET.new $TARGET
+# mv $TARGET.gz $TARGET
```
Comment (eordano) [inline .circleci/hotpatch.sh]:
```suggestion
```

**Pull request merged**